jwt.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/mendersoftware/go-lib-micro/log"
+	"github.com/pkg/errors"
+	"github.com/satori/go.uuid"
+	"io/ioutil"
+	"time"
+)
+
+// token claim names
+const (
+	jwtClaimIssuer     = "iss"
+	jwtClaimSubject    = "sub"
+	jwtClaimExpiration = "exp"
+	jwtClaimTokenId    = "jti"
+)
+
+// JWTHandler jwt generator/verifier
+type JWTHandler interface {
+	GenerateToken(subject string) (*jwt.Token, error)
+	log.ContextLogger
+}
+
+// JWTHandlerRS256 is an RS256-specific JWTHandler
+type JWTHandlerRS256 struct {
+	issuer       string
+	expiresInSec int64
+	privKey      *rsa.PrivateKey
+	log          *log.Logger
+}
+
+func NewJWTHandlerRS256(privKeyPath, issuer string, expiresInSec int64) (*JWTHandlerRS256, error) {
+	priv, err := getRSAPrivKey(privKeyPath)
```

Comment:
Change to `NewJWTHandlerRS256(privKey *rsa.privateKey, issuer string, expiresInSec int64) *JWTHandler`

Shouldn't this accept a logger as well?